src/gdb/gdb.c

```diff
@@ -232,3 +275,19 @@ static void gdb_log_exception(char *message)
 		put_exception_char(*message++);
 
 }
+
+static void write_sr(int sr)
+{
+#ifdef __XTENSA__
```

Comment:
This function should be under arch/xtensa, also plese comment what the assembly does. I'm assuming it writes a byte to a  status register ? If so, the naming and caller semantics should be more generic making it easier to port to non xtensa DSP.

Reply:
`write_sr(int sr)` writes to special register designated by *sr*. I will put some more comments to make it more understandable.
PS: Do you want me to change existing commits or just make another one on top of it which moves xtensa stuff to arch/xtensa/gdb/*?

Reply:
another on top is fine, whatever is easier for you (since the code is not yet functional for everyday users).